apps/backend/listing/tests/createListing.test.ts

```diff
@@ -0,0 +1,117 @@
+import { describe, it, expect, vi } from 'vitest';
+import { createListing } from '../src/createListing';
+import { Request, Response } from 'express';
+import { IDatabase } from 'pg-promise';
+
+describe('Create Listing Endpoint', () => {
+  it('should create a new listing successfully', async () => {
+    const req = {
+      body: {
+        listing: {
+          title: 'Sample Listing',
+          description: 'This is a sample listing description',
+          price: 100,
+          location: {
+            latitude: 48.4284,
+            longitude: -123.3656,
+          },
+          images: [
+            { url: 'https://example.com/image1.png' },
+            { url: 'https://example.com/image2.png' },
+          ],
+        },
+      },
+    } as unknown as Request;
```

Comment:
@Scott-Kenning is there a better way to do type coercion? I hate Javascript

Reply:
Not that I know of, this is why people hate JavaScript. Casting to unknown first makes it type safe (unknown is better than any because.... ???)

Reply:
Actually for this I don't see a reason why it can't just be Request, but `as unknown as Type` is a semantic JS pattern